Microsoft.NET.Build.Containers/ContainerHelpers.cs

```diff
@@ -141,4 +141,64 @@ public static bool NormalizeImageName(string containerImageName, [NotNullWhen(fa
             return false;
         }
     }
+
+    [Flags]
+    public enum ParsePortError
+    {
+        MissingPortNumber,
+        InvalidPortNumber,
+        InvalidPortType,
+        UnknownPortFormat
+    }
+
+    public record ParsePortResult(bool success, Port? port, ParsePortError? parseErrors);
```

Comment:
This was hard to follow for me; in C# I think I'd prefer it outside the type system:

```c#
bool TryParsePort(string input, [NotNullWhen(true)] Port? port, [NotNullWhen(false)] ParsePortError? error)
```

Reply:
refactored to this.